core/src/main/java/org/ldaptive/SingleConnectionFactory.java

```diff
@@ -366,9 +518,30 @@ protected Builder(final Transport transport)
     }
 
 
-    public Builder config(final ConnectionConfig cc)
+    public Builder config(final ConnectionConfig config)
+    {
+      object.setConnectionConfig(config);
+      return this;
+    }
+
+
+    public Builder onOpen(final Function<Connection, Boolean> function)
```

Comment:
These methods seem reasonable in general, but I see that the `SyncReplRunner` use case drove their creation. Can you imagine another use case? Just trying to press on the API change a bit to justify it.

Reply:
I think any use case that involves a persistent search. Content sync isn't the only RFC out there, just the only one we've attempted to use. For instance, AD has the dir sync control.